src/main/java/org/elasticsearch/index/mapper/core/AbstractFieldMapper.java

```diff
@@ -360,6 +368,14 @@ protected String defaultPostingFormat() {
     protected String defaultDocValuesFormat() {
         return null;
     }
+    
+    protected boolean defaultDocValues() {
+        if (writePre2xSettings) {
```

Comment:
Can we rename this variable to something like `indexCreatedBefore20` since it is the way it is used here if I understand correctly?